### ISSUE TITLE
Routes only work in namespace

### ISSUE THREAD

**Issue by zimt28** [CodeAdventure/space-ui#24]
Hey there :)

I'm still using space-ui@3.4.4 or so, I hope you still remember how that version worked ;)
I want to move my app into packages as I think I will get a better structure then. I will not have my global `App` namespace anymore then, so before doing that, I tried to remove it.

All I did was changing this
```coffee
@injector.map('PatientsNewController').toSingleton App.PatientsNewController
# to
@injector.map('PatientsNewController').toSingleton PatientsNewController
```

and 

```coffee
class App.PatientsNewController
# to
class @PatientsNewController
```

I don't know why, but it just stopped working. I get my "not found" page, and the url changes to `undefined`. Do you have an idea what might be wrong?

I can access `PatientsIndexController` from the console and when I log it before mapping it, I can get it there, too. Really strange :)

**Comment by Sanjo** [comment]
In your first code snippet you use PatientsNewController instead of PatientsIndexController.

**Comment by zimt28** [comment]
Oh, I see. But that's just because I rejected my changed and copied the wrong code over as an example :)

**Comment by DominikGuzei** [comment]
Hey @zimt28, when you start splitting up your application into packages (which i highly recommend) then you can start using a `Space.Module` for each package which publishes itself into the global Space environment where it can be required by the main `Space.Application`. Here is some example code:

```coffeescript
Space.Module.extend(function() {
      this.publish(this, 'FirstModule'); # publishes this module into global space environment
      return {
        configure: function() {
          this.injector.map('testValue').toStaticValue(testValue); # application injector
        }
      }
    });

    Space.Application.create({
      RequiredModules: ['FirstModule'], # require your modules which might live in packages
      Dependencies: { testValue: 'testValue' }, # you can also use mapped values from modules
      configure: function () { testResult = this.testValue; }
    });
```

This topic is actually not really related to `space:ui` as the dependency injection and application infrastructure comes from the package [space:base](https://github.com/CodeAdventure/meteor-space) and you can read through [the integration tests](https://github.com/CodeAdventure/meteor-space/tree/master/tests/integration) to get a rough feeling how applications and modules work together.

Long story short: `Space.Application` extends `Space.Module` and has a very similar api. The biggest difference is: There is only a single `Space.Application` in your app with a single `Space.Injector` instance that is automatically shared with all required modules. So you can map dependencies within your modules like you did in your application before and access them from anywhere. One way to organize the dependencies is to namespace them based on the package name, e.g: `@injector.map('MyModule.AwesomeServiceClass').asSingleton()`

**Comment by zimt28** [comment]
Ok cool, thanks for your response :) I have different plans at the moment: Put everything into packages and move to the newest version of space-ui. Maybe switch to meteorhacks flow router and layout as well. The way I'm doing this is to move things into packages *without* space-ui. Things need to keep working and it's much easier for me to move the code without. And I need to rewrite most things for the new space-ui anyway :) However, do you think there's a way I can keep the stores? Like `drop space-ui -> move into packages with stores only -> add new space-ui`? Enjoy your weekend!

**Comment by DominikGuzei** [comment]
@zimt28 ok, I am not sure if I could follow but you could also just split up your code into packages and work with a single `Space.Application` that maps everything for you. The reason why I split packages up into `Space.Module` is that you can have integration tests for each individual package and check if they are doing the right thing when sending commands and publishing events into the system :wink:

**Comment by zimt28** [comment]
Now that's cool! Great, the `find().fetch()[0]` solution was kinda ugly :) I'll try to ask my question differently: What I want to know is just, if it's possible to just use the store from space-ui. The rest was just explanation. Migration to the newest iron:router, space-ui, and moving into packages at once is too much. So I want to put my code into packages without space-ui, but keep the store. So that I can, after all, implement the new version of space-ui :) Makes sense?

**Comment by DominikGuzei** [comment]
Ah ok, now I understand – you mean like just using the store class within your packages? It is possible, but you have to extract the code manually from this Github repository. And of course you would also have to extract parts of the other stuff for the flux messaging etc. so I am not sure what the benefit of that would be?

And to come back to your original question, actually I don't know why it doesn't work – it should! Did you require the package from the main app in `.meteor/packages`?

**Comment by DominikGuzei** [comment]
@zimt28 any updates on that?

**Comment by zimt28** [comment]
Yes :) I'm still busy moving everything into packages, already took some time as I've never created packages before .. Wish I had started with packages :D

I just dropped the old space-ui and copied the old store class so that I don't have to change my stores:

```coffeescript
class Store

  state: new ReactiveVar()

  ready: ->
    @state.set @setInitialState()

  setInitialState: -> {}

  getState: (path) ->
    path = if path? then path.split "." else []
    state = @state.get()
    state = state[key] for key in path
    return state

  setState: (path, value) ->

    # Changes the whole state at once if no path is given
    if arguments.length < 2
      value = path
      @state.set value
      return

    # Sets nested property at path
    existingState = @state.get()
    newState = _.clone existingState

    path = path.split "."
    property = newState

    for name, index in path

      if index + 1 < path.length
        # build up missing object path
        property[name] ?= {}
        property = property[name]
      else
        property[name] = value

    @state.set newState

Store::get = Store::getState
Store::set = Store::setState
```
Works great :) I'm also moving to the newest `iron:router`, and can then get my head around the current space-ui :)

**Comment by DominikGuzei** [comment]
Should I close this thread? :wink: I guess this is no "real issue" anymore
